lib/site_prism/element_container.rb

```diff
@@ -170,16 +170,21 @@ def deduce_iframe_element_selector(selector)
       selector.is_a?(Integer) ? "iframe:nth-of-type(#{selector + 1})" : selector
     end
 
-    def extract_section_options(args, &block)
-      case
-      when args.first.is_a?(Class)
+    def extract_section_options(section_name, args, &block)
+      if args.first.is_a?(Class)
         section_class = args.shift
-      when block_given?
+      elsif block_given?
         section_class = Class.new SitePrism::Section, &block
+      elsif Object.const_defined?(section_class_name(section_name))
+        Object.const_get(section_class_name(section_name))
```

Comment:
This seems like it leaves `section_class` as nil.  Is that intended?

If the goal here is to have `Object.const_get` raise NameError  - then I don't think that will work since this logic won't be called due to the condition above (`Object.const_defined?`).  It will then follow the else and raise your ArgumentError below.  And if that is the intent here it would be nice if it was more explicitly spelled out for the next dev to visit this code.

Reply:
After a bit more consideration I think this whole thing CAN return nil for section_class.  I tried setting up one of the cucumber tests to try out this feature and ended up with a whole lot of:

```
      undefined method `new' for nil:NilClass (NoMethodError)
      ./lib/site_prism/element_container.rb:28:in `block (2 levels) in section'
      ./features/step_definitions/page_section_steps.rb:8:in `/^I can see a section in a section$/'
      features/page_sections.feature:18:in `Then I can see a section in a section'
```

It's possible I'm wrong but I think it's also possible this feature isn't well tested.